.github/actions/entrypoint.ps1

```diff
@@ -0,0 +1,101 @@
+
+$playbooksChanged = (Get-Item env:playbooksChanged).value
+$playbookFilesList = (Get-Item env:playbookFilesList).value
+$mainTemplateOrCreateUiDefinitionTemplateChanged = (Get-Item env:mainTemplateOrCreateUiDefinitionTemplateChanged).value
+
+$isDataConnectorFolderNameWithSpace = (Get-Item env:isDataConnectorFolderNameWithSpace).value
+$dataConnectorFileNames = (Get-Item env:dataConnectorFileNames).value
+$hasDataConnectorFileChanged = (Get-Item env:hasDataConnectorFileChanged).value
+
+Import-Module '/dist/armttk/arm-ttk/arm-ttk.psd1' 
+$MainTemplatePath = './dist/Package'
+
+ls ./
+Write-Host "AA"
+ls ./dist/
+Write-Host "BB"
+ls ./dist/Playbooks
+Write-Host "CC"
+
+# RUN FOR MAINTEMPLATE.JSON FILE
+if ($mainTemplateOrCreateUiDefinitionTemplateChanged -eq $true)
+{
+    Write-Host "Running ARM-TTK on MainTemplate.json and/or CreateUiDefinition.json file, as change is identified those files."
+    $MainTemplateTestResults = Test-AzTemplate -TemplatePath $MainTemplatePath
+    $MainTemplateTestPassed =  $MainTemplateTestResults | Where-Object { -not $_.Failed }
+    Write-Output $MainTemplateTestPassed
+
+    $MainTemplateTestFailures =  $MainTemplateTestResults | Where-Object { -not $_.Passed }
+
+    if ($MainTemplateTestFailures) {
+        Write-Host "CreateUiDefinition.json and/or MainTemplate.json templates did not pass the arm-ttk tests"
+        exit 1
+    } 
+    else {
+        Write-Host "All tests passed for the given files of CreateUiDefinition.json and/or MainTemplate.json!"
+    }
+}
+
+# Data Connector file change
+if ($hasDataConnectorFileChanged -eq $true)
+{
+    Write-Host "Running ARM-TTK on Data Connectors Folder, '$dataConnectorFileNames' files, as change is identified those file"
+    $dataConnectorFolderName = "Data Connectors"
+    if($isDataConnectorFolderNameWithSpace -ne $true)
+    {
+        $dataConnectorFolderName = "DataConnectors"
+    }
+
+    $dataConnectorFilesList = $dataConnectorFileNames.Split(" ")
+    foreach($dataConnectorFileItem in $dataConnectorFilesList)
```

Comment:
Please check if you give the file names in an array to the test-aztemplate command to run all files at once insead of looping every file and giving the results

Reply:
I have tried below things:
1. Copy all files changed in PR to a new folder and run arm-ttk on that folder. The problem I see here is if we have multiple files with same name eg: In Playbooks folder we can have multiple sub-folders with same name to file say azureDeploy.json then this will fail with an unexpected error of duplicate file which already exist in the destination folder.
2. If we copy folders of data connector/playbooks etc folder and also has change for maintemplate/createui then the result provided by ARM-TTL is not showing properly as shown below and it also doesnt provide any details of which file the test were ran for.
![image](https://user-images.githubusercontent.com/107389644/210979267-e074578b-7a46-4259-98db-3235c54148b0.png)

As per email communication with Prateek. He is specifying if we can get details on which file ARM-TTK was running for. So my recommendation here is that we should run ARM-TTK separetely for each file so that it will have readablility and easy to identify and fix any ARM-TTK issues in tests.